### PR TITLE
Main and authorization pages

### PR DESCRIPTION
**Pull request by Andrew77747**

### CODE REVIEW

File: Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs [hunk 1]
```diff
@@ -8,5 +9,9 @@ namespace WebShop.Tricentis.Framework.PageObject
 {
     class AuthorizationPage
     {
+        private readonly By _inputEmail = By.CssSelector("[name='Email']");
+        private readonly By _inputPassword = By.CssSelector("[name='Password']");
+        private readonly By _chekboxRememberMe = By.CssSelector("#RememberMe");
+        private readonly By _loginButton = By.CssSelector("[class='button-1 login-button']");
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs]: Please, change [class=button-1 to .button-1
also, please try find more beauty selector

**Review by IuriiBaikov — CHANGES_REQUESTED** [review]

File: Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs (hunk 1 above)
Comment (Andrew77747) [reply Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs]: Changed

File: Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs [hunk 2]
```diff
@@ -1,12 +1,53 @@
...
+        private readonly By _chekboxRememberMe = By.CssSelector("#RememberMe");
+        private readonly By _loginButton = By.CssSelector(".login-button");
+        private readonly By _userName = By.CssSelector(".header - .account");
+        private readonly By _logOutButton = By.CssSelector(".ico-logout");
+
+
+        private readonly IWebDriver _driver;
+
+        public AuthorizationPage(IWebDriver driver)
+        {
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs:18]: Please, do the same as in ManePage.cs.

File: Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs [hunk 3]
```diff
@@ -1,12 +1,53 @@
...
     {
+        private readonly By _inputEmail = By.CssSelector(".email");
+        private readonly By _inputPassword = By.CssSelector(".password");
+        private readonly By _chekboxRememberMe = By.CssSelector("#RememberMe");
+        private readonly By _loginButton = By.CssSelector(".login-button");
+        private readonly By _userName = By.CssSelector(".header - .account");
+        private readonly By _logOutButton = By.CssSelector(".ico-logout");
+
+
+        private readonly IWebDriver _driver;
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Framework/PageObject/AuthorizationPage.cs:15]: Please add inheritance from BasePage

File: Solution of WebShop/WebShop.Tricentis.Framework/PageObject/MainPage.cs [hunk 4]
```diff
@@ -1,17 +1,52 @@
...
+        {
+            _driver.FindElement(_login).Click();
+        }
+
+        public string GetUrl()
+        {
+            return _driver.Url;
+        }
+
+        public void TypeRequest()
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Framework/PageObject/MainPage.cs]: Please rename to Search

File: Solution of WebShop/WebShop.Tricentis.Framework/PageObject/MainPage.cs [hunk 5]
```diff
@@ -1,17 +1,52 @@
...
+        {
+            _driver.FindElement(_inputSearch).SendKeys("Simple computer");
+        }
+
+        public void ClickSearch()
+        {
+            _driver.FindElement(_searchButton).Click();
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Framework/PageObject/MainPage.cs:40]: Please mooove it to Search() method

File: Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/Authorization.feature [hunk 6]
```diff
@@ -0,0 +1,29 @@
+﻿Feature: Authorization
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/Authorization.feature:1]: Please add formating in this file

File: Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/AuthorizationSteps.cs [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            _mainPage.ClickSearch();
+        }
+
+        [When(@"I click the found item")]
+        public void WhenIClickTheFoundItem()
+        {
+            _mainPage.ClickFoundItem();
+        }
+
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/AuthorizationSteps.cs]: Please remove superline

File: Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/Authorization.feature [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+Scenario: Check authorization page URL
+	Given I'm on the main page
+	When I click login 
+	Then I'm on the authorization page
+
+Scenario: Check LogIn
+   Given I go to the authorization page
+   When I enter login
+   And I enter password
+   And I click login the LogIn button
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/Authorization.feature]: Please check is it right?

File: Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/Authorization.feature [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+   Given I'am registered  
+   When I click logout
+   Then I'm on the main page
+   And I don't see 'andrew-walker@yandex.ru'
+   And I don't see 'Log out'
+
+   Scenario: Check the search input
+   Given I'm on the main page
+   When I type my request
+   And I click the search button
```
Comment (IuriiBaikov) [inline Solution of WebShop/WebShop.Tricentis.Tests/Scenarios/Authorization.feature]: I believe it can be joined

**Review by IuriiBaikov — CHANGES_REQUESTED** [review]

**Review by IuriiBaikov — APPROVED** [review]

**Pull request merged**